### PR TITLE
Add lsp support for purescript

### PR DESCRIPTION
**Pull request by nobv**
I made PR because there are nothing lsp support for purescript.
But I'm not sure if this change is enough.
I am waiting for your opinions.
Thanks!

### CODE REVIEW

File: layers/+lang/purescript/funcs.el
Comment (smile13241324) [inline layers/+lang/purescript/funcs.el:1]: This header should say purescript layer. Also you should be named as the author

File: layers/+lang/purescript/layers.el
Comment (smile13241324) [inline layers/+lang/purescript/layers.el:12]: Removing the dependency on the node layer will probably break part of the layer code, please keep this.

**Comment by nobv** [comment]
@smile13241324 
Thank you for your comments!
I'm testing now with updated codes, but I have a problem.
When open file first time, lsp-ui doesn't work...
But after that and open other file or restart lsp, lsp-ui does.
I don't know why...
Could you some advice please?

File: layers/+lang/purescript/packages.el
Comment (smile13241324) [inline layers/+lang/purescript/packages.el:37]: You should still defer loading of this mode.

File: layers/+lang/purescript/packages.el
Comment (smile13241324) [inline layers/+lang/purescript/packages.el:39]: I would add this to `purescript-mode-hook` instead of the local vars hook, I am not sure that local vars hook is also executed when the buffer is restored from a previous editing session.

File: layers/+lang/purescript/funcs.el
Comment (smile13241324) [inline layers/+lang/purescript/funcs.el:47]: This is wrong, you need to check for the value of the selected backend, however this has already been done upstream, so here you only need to setup company unconditionally. The check for the lsp layer should be removed.

Also you don't need a `progǹ` in a functions context, please remove it for it consumes resources at run time.

Ahh and please inline this function in `spacemacs//purescript-setup-company`.

File: layers/+lang/purescript/funcs.el
Comment (smile13241324) [inline layers/+lang/purescript/funcs.el:39]: Same here, please remove this function and inline the resulting call in `spacemacs//purescript-setup-backend`.
You only need `(lsp)` the rest can be removed.

**Review by smile13241324 — CHANGES_REQUESTED** [review]
I have had another look at your PR and probably found the reason why it is not working properly on your side. Also I have attached some general comments about how to simplify your code.

Please check if this solves your issue and ping me when I should have another look.
I am not entirely sure that the issue you are seeing has anything to do with your changes there have been a lot of changes on LSP mode lately this could very well be an issue caused by this rather then by your code.

**Comment by smile13241324** [comment]
Btw, looks like CI failed, can you rebase the latest develop on your branch to make sure you also have the latest bug fixes available, I have fixed some LSP related stuff just before 2 days, maybe your issue is gone already.

**Comment by JAremko** [comment]
Don't mind `windows-latest, 26.3, core` fail. Its something that we have to fix.

**Comment by nobv** [comment]
@smile13241324 
As you said, I could confirm LSP mode work well after rebase.
Thanks for your all support!

**Comment by smile13241324** [comment]
Great, so that means you are finished, shall I have another review then?

**Comment by nobv** [comment]
@smile13241324 
Yes, please!

File: layers/+lang/purescript/README.org
Comment (smile13241324) [inline layers/+lang/purescript/README.org:65]: typo, ....once a purescript file....

File: layers/+lang/purescript/README.org
Comment (smile13241324) [inline layers/+lang/purescript/README.org:89]: Too much blank lines here, please reduce to one.

**Review by smile13241324 — CHANGES_REQUESTED** [review]
Ups just noticed that you have merged a new version of develop on your local branch. Unfortunately I cannot merge it this way please:
* Reset the branch to before the merge
* Then rebase develop on top of it
* Then cherry pick your last commit
* Fix the few open points
* Finally please squash your commits into one.

With this done I can merge it.

**Comment by nobv** [comment]
@smile13241324 
Do I need to resolve the error?

**Comment by JAremko** [comment]
@nobv  rebase should help with CI.

**Comment by nobv** [comment]
@JAremko 
I rebased and CI all passed.
Thank you for your replying!

**Review by smile13241324 — APPROVED** [review]
Now it looks good, thanks for all the hard work you have put into this @nobv.

**Comment by smile13241324** [comment]
Thank you for contributing to spacemacs 💜, I have cherry picked your commit into develop.

**Pull request merged**
